### PR TITLE
fix(iOS): ListViewItem detaching from its Parent

### LINKED ISSUE #13199: [iOS] Losing reference to datacontext of an element outside of ItemTemplate on CollectionChanged
### Current behavior

actual

```xaml
<Page.Resources>
		<DataTemplate x:Key="ItemTemplate">
			<StackPanel Background="LightBlue">
				<TextBlock>
					<Run Text="Item content:" />
					<Run Text="{Binding Name}" />
				</TextBlock>

				<TextBlock>
					<Run Text="Parent data context:" />
					<Run Text="{Binding DataContext, ElementName=Parent}" />
				</TextBlock>

				<Button Command="{Binding DataContext.Command, ElementName=Parent}"
						CommandParameter="{Binding}"
						Content="Click" />
			</StackPanel>
		</DataTemplate>
	</Page.Resources>

	<Grid Margin="8, 32">

		<ListView ItemsSource="{Binding Items}"
				  x:Name="Parent"
				  ItemTemplate="{StaticResource ItemTemplate}" />
	</Grid>
```

When updating the collection bound to `ListView`, the `ItemTemplate` loses the DataContext of `Parent`.

### Expected behavior

act

On android/winui, the `Parent` `DataContext` reference is never lost.

### How to reproduce it (as minimally and precisely as possible)

- Download this sample [ElementNameBinding.zip](https://github.com/unoplatform/uno/files/12317267/ElementNameBinding.zip)
- Deploy on iOS
- Click on "Click" button on each item

### Workaround

A potential workaround is to refrain from referencing elements outside of the ItemTemplate itself. For example, you could pass the parent command for each item manually.

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.0.0-dev.1923

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

17.6.1

### Relevant plugins

_No response_

### Anything else we need to know?

The bug is not happening on uno.winui 4.7.37

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #13199 

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
When removing and adding a new Item in a ListView, the new items lose the connection with their Parent.

### <samp>🤖 Generated by Copilot at 1928530</samp>

This pull request adds a new sample control to demonstrate data context and commands in a `ListView`, and fixes a bug in the `ListViewBaseSource` class on iOS that caused null parent issues for reused cells. The sample control uses a user control, a view model, and a generic command class to show how to bind data and execute actions in a `ListView` item template. The bug fix adds a condition to check and set the parent of the selector item in the `GetCell` method.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			if (item is null)
+			{
+				Console.WriteLine("Item was null");
+				return;
+			}
+
+			var newItem = item with { Name = $"Selected item {item.Index}" };
+
+			Items.Remove(item);
+			Items.Insert(newItem.Index, newItem);
```
Comment (Xiaoy312) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs:58]: this should be verifiable under runtime test, no?

Comment (ajpinedam) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs:58]: Still trying to figure it out. Once we Remove then Insert a new Item, we need to access the nth-ListViewitem to validate if the `GetParent()` method returns other than `null.`

Comment (jeromelaban) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs:58]: You'll likely need to scroll to the required position. That should be working properly.

**Comment by jeromelaban** [comment]
You can also restart the build when done, the build issues were temporary.

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs (hunk 1 above)
Comment (ajpinedam) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs:58]: Added Unit test validating if the container's parent for the added item is not null. This is the behaviour I tested manually when finding the fix.

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@ajpinedam its still failing on some member issue:
```
src\SamplesApp\SamplesApp.UWP\obj\x86\Release\Windows_UI_Xaml_Controls\ListView\ListView_DataContext_Reference.g.i.cs
(18,59): Error CS0108: 'ListView_DataContext_Reference.Parent' hides inherited member 
'FrameworkElement.Parent'. Use the new keyword if hiding was intended.
```

**Pull request merged**